test/testcafe/framework/page-objects/EnrollEmailPageObject.js

```diff
@@ -21,12 +22,55 @@ export default class EnrollAuthenticatorPhonePageObject extends BasePageObject {
   }
 
   async enterCodeFromEmailLinkExists() {
-    const elCount = await Selector(ENTER_CODE_FROM_EMAIL).count;
-    return elCount === 1;
+    if (userVariables.v3) {
+      const elCount = await Selector('[type="button"]').count;
+      return elCount === 1;
+    } else {
+      const elCount = await Selector(ENTER_CODE_FROM_EMAIL_CLASS).count;
+      return elCount === 1;
+    }
   }
 
   async clickElement(selector) {
     await this.form.clickElement(selector);
   }
 
+  async clickEnterCodeInstead() {
+    if (userVariables.v3) {
+      await this.form.clickButton(ENTER_CODE_FROM_EMAIL_TEXT);
+    } else {
+      await this.form.clickElement(ENTER_CODE_FROM_EMAIL_CLASS);
+    }
+  }
+
+  async resendEmailExists() {
+    if (userVariables.v3) {
+      return Selector('[role="alert"]').exists;
+    }
+
+    const isHidden = await Selector(RESEND_EMAIL).hasClass('hide')
+    return !isHidden;
+  }
+
+  resendEmailText() {
+    if (userVariables.v3) {
+      return this.form.el.find('[role="alert"]').textContent;
+    }
+      return this.form.el.find(RESEND_EMAIL).textContent;
+  }
+
+  async clickResendEmail() {
+    if (userVariables.v3) {
+      const resendEmail = this.form.getLink('Send again');
+      await this.t.click(resendEmail);
+    }
+    else {
+      const resendEmailLink = this.form.el.find(RESEND_LINK);
```

Comment:
Are we able to use the same `getLink` code as in v3 to pick this link out? As long as both are an `a` tag with 'Send again' as the text it should work for both.

Reply:
`this.form.getLink` looks for `role="link"`. Even though the v2 element has an `a` tag, it doesn't have `role="link"` and `getLink` fails to pick it up